src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs

```diff
@@ -285,5 +296,69 @@ private static IEnumerable<Reference> GetProjectReferences(EnvDTEProject project
                 }
             }
         }
+
+        /// <summary>
+        /// Get the unique names of all references which have ReferenceOutputAssembly set to false.
+        /// </summary>
+        private static async Task<List<string>> GetExcludedReferences(
+            EnvDTEProject project,
+            IVsEnumHierarchyItemsFactory itemsFactory)
+        {
+            var excludedReferences = new List<string>();
+
+            var hierarchy = VsHierarchyUtility.ToVsHierarchy(project);
+            var buildPropertyStorage = hierarchy as IVsBuildPropertyStorage;
+
+            // Get all items in the hierarchy, this includes project references, files, and everything else.
+            IEnumHierarchyItems items;
+            if (ErrorHandler.Succeeded(itemsFactory.EnumHierarchyItems(
+                hierarchy,
+                (uint)__VSEHI.VSEHI_Leaf,
+                (uint)VSConstants.VSITEMID.Root,
+                out items)))
+            {
+                // Loop through all items
+                uint fetched;
+                VSITEMSELECTION[] item = new VSITEMSELECTION[1];
+                while (ErrorHandler.Succeeded(items.Next(1, item, out fetched)) && fetched == 1)
+                {
+                    // Check if the item has ReferenceOutputAssembly
+                    string value;
+                    if (ErrorHandler.Succeeded(buildPropertyStorage.GetItemAttribute(
```

Comment:
Earlier `buildPropertyStorage` is created via an `as` cast which indicates it can be `null`.  Won't this NRE when the cast fails? 

If the cast is always expected to succeed I would just use a normal `(IVsBuildPropertyStorage)` cast to indicate that failure is not expected.